### ISSUE TITLE
Adding content: none to paket.dependencies && paket install does not remove installed content

### ISSUE THREAD

**Issue by agross** [fsprojects/Paket#687]
I had this in my depfile:
```
source https://www.nuget.org/api/v2

nuget JasmineTest 2.0.0 
```

I ran `paket install` and had the content added. I then added `content: none` to the depfile, executed `git clean -xdf` and ran `pakat install` again. The new `content: none` option was not picked up and the content was restored.

**Comment by matthid** [comment]
I can confirm that this is happening, however I was never able to repro it...

**Comment by agross** [comment]
I think it might be related to detecting whether the depfile has changed between paket installs. Just a guess, though. 




Alex
-- 
Alexander Groß
Tiny phone, tiny mail

**Comment by theimowski** [comment]
Will investigate, hopefully soon

**Comment by theimowski** [comment]
@agross I can't reproduce - you sure `content: none` was spelled correctly?

**Comment by agross** [comment]
Here's a reproduction: https://github.com/agross/content-repro

Uncomment the commented line in https://github.com/agross/content-repro/blob/master/paket.dependencies and remove the original nuget. Then run `paket install` and `git status` after that.

Here's my output:
```bash
$ paket install
Paket version 0.35.0.0
found: content-repro\paket.dependencies
Resolving packages:
 - JasmineTest 2.2.0
Locked version resolutions written to content-repro\paket.lock
1 second - ready.

$ git diff
diff --git a/paket.dependencies b/paket.dependencies
index 5f4c593..0e17d56 100644
--- a/paket.dependencies
+++ b/paket.dependencies
@@ -1,4 +1,4 @@
 source https://www.nuget.org/api/v2

-nuget JasmineTest
-# nuget JasmineTest content: none
+#nuget JasmineTest
+nuget JasmineTest content: none
```

**Comment by theimowski** [comment]
I don't think we support this option on package level, but rather on global level. See http://fsprojects.github.io/Paket/dependencies-file.html#No-content-option

**Comment by agross** [comment]
@forki added it on all levels: https://github.com/fsprojects/Paket/issues/587

**Comment by theimowski** [comment]
Aww sorry wasn't aware of that...

**Comment by agross** [comment]
No worries :-)

**Comment by ctaggart** [comment]
This bug applied to both 0.33.13 and 0.35.0 that I tested. It only fails when no version is specified. These all work:

```
nuget elmah 1.2.2 content: none
nuget Microsoft.SqlServer.Types 11.0.0 content: none
nuget Microsoft.ApplicationInsights.Web pre content: none
```

These do not:
```
nuget elmah content: none
nuget Microsoft.SqlServer.Types content: none
nuget Microsoft.ApplicationInsights.Web content: none
```

**Comment by theimowski** [comment]
ok so I investigated a bit more, and the key point here is that you should use ``paket update`` rather than ``paket install``. 
As per the docs:

    install: Ensures that all dependencies in your paket.dependencies file are present in the `packages` directory and referenced correctly in all projects.

while

    update: Recomputes the dependency resolution, updates the paket.lock file and propagates any resulting package changes into all project files referencing updated packages.

In other words, if you don't ``paket update``, the lock file won't be affected and ``content: none`` won't be applied

**Comment by agross** [comment]
Hm... I was under the impression @forki changed the model to be more like bundler: Change something in your depfile, run `install` and be done with it.

**Comment by forki** [comment]
"more like" is correct. But it doesn't detect all changes, yet

**Comment by birryree** [comment]
I was definitely affected by this, but what happens depends on what type of option you set (global or per-package).

* Changing a global option and then running `paket install` will regenerate the `paket.lock` file with that option set.

* Changing a per-package option (like adding `content: none`) and running `paket install` will not regenerate the `paket.lock` with that option set on that package.

* Adding a new package and running `paket install` will also not update any changes that reflect per-package options, but will reflect global option changes.

The workarounds/solutions are to either remove `paket.lock` and re-run (if possible) or run `paket update`.
